### PR TITLE
cake-5067 | Added isSandbox function to react common

### PR DESCRIPTION
**Pull request by cschucode**
Wanted a check for sandbox environment that could be reused across different apps. This function currently checks if it's on a sandbox environment by looking for properties on the window object. Currently can be used to check on:
- fepo
- mediawiki
- f2
- upstream

### CODE REVIEW

File: source/utils/environment.js [hunk 1]
```diff
@@ -22,3 +24,23 @@ export function getServicesBaseURL() {
 export function getEventLoggerBase() {
     return `${getServicesBaseURL()}event-logger`;
 }
+
+export function isSandbox() {
+    if (!isBrowser() || isProduction()) {
+        return false;
+    }
+    // fepo
+    if (isBrowser() && get(window, 'window.fandom.environment.environment', '').indexOf('sandbox') > -1) {
```
Comment (mklucsarits) [inline source/utils/environment.js]: You don't need to check `isBrowser` for each of these; the function will have already returned `false` above when checking `!isBrowser()`.

Comment (cschucode) [reply source/utils/environment.js]: Good point. I had a couple of theories in my head accidentally combined the two. Good catch.

File: source/utils/environment.js [hunk 2]
```diff
@@ -22,3 +24,23 @@ export function getServicesBaseURL() {
 export function getEventLoggerBase() {
     return `${getServicesBaseURL()}event-logger`;
 }
+
+export function isSandbox() {
+    if (!isBrowser() || isProduction()) {
```
Comment (JoshuaRogan) [inline source/utils/environment.js]: I think we need to remove the check for production here. We will probably want to include the check for `isSandbox ` inside of `isProduction`. With this code, we will end up with a circular dependency causing an infinite loop.

File: source/utils/environment.spec.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
+import { isSandbox } from './environment';
+
+// Unit tests for isSandbox() in multiple environments
+describe('Tests for is Sandbox: ', () => {
+    test('It should return false when window object property is undefined', () => {
+        expect(isSandbox()).toEqual(false);
+    });
+
+    test('It should return false if on production', () => {
```
Comment (JoshuaRogan) [inline source/utils/environment.spec.js]: I believe this is for f2.

File: source/utils/environment.spec.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        global.window = {
+            wgTransactionContext: {
+                env: 'sandbox',
+            },
+        };
+
+        expect(isSandbox()).toEqual(true);
+    });
+
+    test('It should return true if sandbox is on f2', () => {
```
Comment (cschucode) [inline source/utils/environment.spec.js]: Test for f2 when true.

File: source/utils/environment.spec.js (hunk 3 above)
Comment (cschucode) [reply source/utils/environment.spec.js]: See below.

Comment (cschucode) [reply source/utils/environment.spec.js]: In there already.

File: source/utils/environment.js [hunk 5]
```diff
@@ -22,3 +24,23 @@ export function getServicesBaseURL() {
...
+    if (get(window, 'window.fandom.environment.environment', '').indexOf('sandbox') > -1) {
+        return true;
+    }
+    // mw
+    if (get(window, 'window.wgTransactionContext.env', '').indexOf('sandbox') > -1) {
+        return true;
+    }
+    // f2
+    if (get(window, 'window.fandom.config.environment.env', '').indexOf('sandbox') > -1) {
+        return true;
```
[the remark this answers is not in the document]
Comment (cschucode) [reply source/utils/environment.js]: Added check for upstream and updated tests.

**Pull request merged**